src/inventory_ui.cpp

```diff
@@ -401,14 +454,329 @@ void inventory_selector::print_right_column() const
+        nc_color invlet_color = (a == selected) ? selected_line_color :  // If selected, selected_line_color, otherwise (if hotkey assigned, yellow, otherwise white) 
+                (g->u.assigned_invlet.count(it.invlet) ? c_yellow : c_white);
+        nc_color name_color = (a == selected) ? selected_line_color : it.color_in_inventory();
+        
+        // Render the cell
+        if(curr_line >= 3 && curr_line < TERMY) {  // Top line
+        
+            // Cell Border Stuff
+            if(row_i == 0) {
+                mvwputch(w_inv, curr_line, 1, c_white, cat_continued ? LINE_XXXO : LINE_OXXO);
+            } else {
+                mvwputch(w_inv, curr_line, 1 + (row_i * 25), c_white, cat_continued ? LINE_XXXX : LINE_OXXX);
+            }
+            if(!cat_continued) { // This is a continuation, no need to render top line
+                for(char x = 2; x < 26; x++) {
+                    mvwputch(w_inv, curr_line, x + (row_i * 25), c_white, LINE_OXOX);
+                }
+                mvwputch(w_inv, curr_line, 26 + (row_i * 25), c_white, LINE_OOXX);
+            } else { // Just edit the top right corner
+                mvwputch(w_inv, curr_line, 26 + (row_i * 25), c_white, LINE_XOXX);
+            }
+            // End Cell Border Stuff
+        }
+        
+        std::string item_name = it.label(); // Get object's display name
+        size_t count = 1;
+        if(cur_entry.slice != NULL) {
+            count = cur_entry.slice->size();
+            if(count > 1) {
+                item_name = it.label(count);
+            }
+        }
+        
+        if(curr_line >= 2 && curr_line + 1 < TERMY) {  // Second line
+            // Cell Border Stuff
+            if(row_i == 0) {
+                mvwputch(w_inv, curr_line + 1, 1, c_white, LINE_XOXO);
+            }
+            for(char x = 2; x < 26; x++) {
+                mvwputch(w_inv, curr_line + 1, x + (row_i * 25), fill, ' ');
+            }
+            mvwputch(w_inv, curr_line + 1, 26 + (row_i * 25), c_white, LINE_XOXO);
+            // End Cell Border Stuff
+        
+            if(it.invlet != 0) { // Show item hotkey (if any)
+                mvwputch(w_inv, curr_line + 1, 2 + (row_i * 25), invlet_color, it.invlet);
+            }
+            
+            if(count > 1) { // Show item count (if more than one)
+                mvwprintz(w_inv, curr_line + 1, 3 + (row_i * 25), fill, "%22ux", count);
+            }
+            
+            if(OPTIONS["ITEM_HEALTH_BAR"] && (it.damage != 0 || it.is_armor()) && !it.is_null()) { // Show item damage (if appropriate)
+                auto const &nc_text = get_item_hp_bar(it.damage);
+                mvwprintz(w_inv, curr_line + 1, 6 + (row_i * 25), nc_text.second, nc_text.first.c_str());
+            }
+            
+            mvwprintz(w_inv, curr_line + 1, 10 + (row_i * 25), fill, it.get_mod_string().c_str()); // Show mod count (if any)
+        }
+        
+        std::string temp[2];
```

Comment:
Consider using `std::array<std::string, 2>` as type instead of a raw array. `split_names` should accept that one instead of raw pointer. When looking at `split_names`, one can't see what the pointer is supposed to be, no documentation that it should point to an array with at least 2 elements. Using `std::array` documents that requirement and triggers a compiler error if the argument don't match that requirement.